### PR TITLE
Bumps niftany and fixes things for new :cop: in town

### PR DESCRIPTION
**Pull request by cdmo**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -38,7 +38,6 @@ group :development, :test do
   gem 'pry-byebug'
   gem 'rails-controller-testing'
   gem 'rspec-rails'
-  gem 'rubocop'
```
Comment (cdmo) [inline Gemfile:41]: no need to specify this it comes with niftany

**Comment by cdmo** [comment]
Citations issues by 👮‍♂️  that were ameliorated:

```
➜  psulib_blacklight git:(dependency-bump) bundle exec rubocop  -a
.rubocop.yml: Metrics/LineLength has the wrong namespace - should be Layout
The following cops were added to RuboCop, but are not configured. Please set Enabled to either `true` or `false` in your `.rubocop.yml` file:
 - Layout/EmptyLinesAroundAttributeAccessor (0.83)
 - Style/SlicingWithRange (0.83)
For more information: https://docs.rubocop.org/en/latest/versioning/
Inspecting 72 files
............................C........................CCCC.......C.......

Offenses:

config/environments/development.rb:25:6: C: Rails/FilePath: Please use Rails.root.join('path/to') instead.
  if Rails.root.join('tmp', 'caching-dev.txt').exist?
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/blackcat.rake:9:5: C: Rails/RakeEnvironment: Include :environment task as a dependency for all Rake tasks.
    task :index do
    ^^^^^^^^^^^
lib/tasks/blackcat.rake:17:5: C: Rails/RakeEnvironment: Include :environment task as a dependency for all Rake tasks.
    task :deindex do
    ^^^^^^^^^^^^^
lib/tasks/blackcat.rake:26:5: C: Rails/RakeEnvironment: Include :environment task as a dependency for all Rake tasks.
    task :create_fixtures do
    ^^^^^^^^^^^^^^^^^^^^^
lib/tasks/blackcat.rake:28:22: C: Rails/FilePath: Please use Rails.root.join('path/to') instead.
      traject_path = Rails.root.join('..', 'psulib_traject')
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/blackcat.rake:29:18: C: Rails/FilePath: Please use Rails.root.join('path/to') instead.
      fixtures = Rails.root.join('spec', 'fixtures', 'current_fixtures.json')
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/blackcat.rake:30:19: C: Rails/FilePath: Please use Rails.root.join('path/to') instead.
      marc_file = Rails.root.join('solr', 'sample_data', 'sample_psucat.mrc')
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/docker.rake:6:3: C: Rails/RakeEnvironment: Include :environment task as a dependency for all Rake tasks.
  task :up do
  ^^^^^^^^
lib/tasks/docker.rake:25:3: C: Rails/RakeEnvironment: Include :environment task as a dependency for all Rake tasks.
  task :clean do
  ^^^^^^^^^^^
lib/tasks/docker.rake:31:3: C: Rails/RakeEnvironment: Include :environment task as a dependency for all Rake tasks.
  task :up_zk_config do
  ^^^^^^^^^^^^^^^^^^
lib/tasks/docker.rake:35:3: C: Rails/RakeEnvironment: Include :environment task as a dependency for all Rake tasks.
  task :create_collection do
  ^^^^^^^^^^^^^^^^^^^^^^^
lib/tasks/docker.rake:43:3: C: Rails/RakeEnvironment: Include :environment task as a dependency for all Rake tasks.
  1 ---
  task :run do
  ^^^^^^^^^
lib/tasks/docker.rake:59:3: C: Rails/RakeEnvironment: Include :environment task as a dependency for all Rake tasks.
  task :run_ci do
  ^^^^^^^^^^^^
lib/tasks/docker.rake:71:3: C: Rails/RakeEnvironment: Include :environment task as a dependency for all Rake tasks.
  task :pull do
  ^^^^^^^^^^
lib/tasks/docker.rake:75:3: C: Rails/RakeEnvironment: Include :environment task as a dependency for all Rake tasks.
  task :start do
  ^^^^^^^^^^^
lib/tasks/docker.rake:84:3: C: Rails/RakeEnvironment: Include :environment task as a dependency for all Rake tasks.
  task :down do
  ^^^^^^^^^^
lib/tasks/docker.rake:88:3: C: Rails/RakeEnvironment: Include :environment task as a dependency for all Rake tasks.
  task :ps do
  ^^^^^^^^
lib/tasks/server.rake:4:1: C: Rails/RakeEnvironment: Include :environment task as a dependency for all Rake tasks.
task :ci do
^^^^^^^^
lib/tasks/yarn.rake:5:3: C: Rails/RakeEnvironment: Include :environment task as a dependency for all Rake tasks.
  task :install do
  ^^^^^^^^^^^^^
spec/helpers/catalog_helper_spec.rb:30:3: C: RSpec/LeakyConstantDeclaration: Stub constant instead of declaring explicitly.
  SEPARATOR = '—'
  ^^^^^^^^^^^^^^^
```

Then after rebase got pulled over a few more times:

```
➜  psulib_blacklight git:(dependency-bump) bundle exec rubocop  -a
Inspecting 74 files
....../Users/cdm32/Projects/ruby/psulib_blacklight/app/controllers/etas_controller.rb: Warning: no department given for MethodLength. Run `rubocop -a --only Migration/DepartmentName` to fix.
/Users/cdm32/Projects/ruby/psulib_blacklight/app/controllers/etas_controller.rb: Warning: no department given for MethodLength. Run `rubocop -a --only Migration/DepartmentName` to fix.
W....................................................C.C............

Offenses:

app/controllers/etas_controller.rb:42:5: C: [Corrected] Layout/EmptyLineBetweenDefs: Use empty lines between method definitions.
    def hathitrust_api_response
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/controllers/etas_controller.rb:43:5: W: [Corrected] Lint/RedundantCopDisableDirective: Unnecessary disabling of Metrics/MethodLength.
    # rubocop:disable Metrics/MethodLength
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/controllers/etas_controller.rb:43:23: C: [Corrected] Migration/DepartmentName: Department name is missing.
    # rubocop:disable MethodLength
                      ^^^^^^^^^^^^
app/controllers/etas_controller.rb:57:22: W: [Corrected] Lint/RedundantCopEnableDirective: Unnecessary enabling of Metrics/MethodLength.
    # rubocop:enable Metrics/MethodLength
                     ^^^^^^^^^^^^^^^^^^^^
app/controllers/etas_controller.rb:58:1: C: [Corrected] Layout/TrailingWhitespace: Trailing whitespace detected.
app/controllers/etas_controller.rb:59:22: C: [Corrected] Migration/DepartmentName: Department name is missing.
    # rubocop:enable MethodLength
                     ^^^^^^^^^^^^
app/controllers/etas_controller.rb:60:5: C: [Corrected] Layout/EmptyLineBetweenDefs: Use empty lines between method definitions.
    def hathitrust_uri
    ^^^^^^^^^^^^^^^^^^
spec/controllers/etas_controller_spec.rb:11:5: C: [Corrected] RSpec/EmptyLineAfterHook: Add an empty line after before.
    end
    ^^^
spec/features/ask_a_librarian_spec.rb:22:33: C: [Corrected] Layout/ArgumentAlignment: Align the arguments of a method call if they span more than one line.
                                text: 'Ethical and Social Issues in the Information Age by Joseph Migga Kizza', ...
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

erb-lint 👮‍♂️ nearly hauled me off to the clink for these:

```
Linting 41 files with 12 linters...

Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
In file: app/views/advanced/_advanced_search_facets_as_select.html.erb:11

Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
In file: app/views/advanced/_advanced_search_facets_as_select.html.erb:12

Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
In file: app/views/advanced/_advanced_search_facets_as_select.html.erb:13

Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
In file: app/views/advanced/_advanced_search_facets_as_select.html.erb:14

Layout/LeadingEmptyLines: Unnecessary blank line at the beginning of the source.
In file: app/views/catalog/_index_availability.html.erb:2

Layout/LeadingEmptyLines: Unnecessary blank line at the beginning of the source.
In file: app/views/catalog/_show_availability.html.erb:2

Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
In file: app/views/catalog/_search_form.html.erb:12

7 error(s) were found in ERB files
ERBLint FAILED!
```
and more after rebase:

```
Linting 42 files with 12 linters...

Layout/ArgumentAlignment: Align the arguments of a method call if they span more than one line.
In file: app/views/hathitrust/etas.html.erb:9

Layout/ArgumentAlignment: Align the arguments of a method call if they span more than one line.
In file: app/views/hathitrust/etas.html.erb:7
```

Think we've made amends, should be good now.

One last thing is to get rid of the warning rubo gives us about cops that need config that I think will be addressed by https://github.com/psu-libraries/niftany/pull/16

File: app/helpers/facets_helper.rb [hunk 2]
```diff
@@ -30,7 +30,7 @@ def pivot_facet_child_in_params?(field_name, item, pivot_in_params = false)
   end
 
   def pivot_facet_in_params?(field_name, item)
-    field_name = item.field if item && item.respond_to?(:field)
+    field_name = item.field if item&.respond_to?(:field)
```
Comment (banukutlu) [inline app/helpers/facets_helper.rb:33]: nice 👍

File: spec/helpers/catalog_helper_spec.rb [hunk 3]
```diff
@@ -19,21 +19,22 @@
...
 
-  SEPARATOR = '—'
   describe '#subjectify' do
     let (:field_data) { ['Renewable energy sources—Research—United States—Finance—History',
                          'Federal aid to research—United States—History'] }
     let (:subjects_doc) { { value: field_data } }
 
     context 'when subjects include subfields v, x, y, and z' do
       it 'provides links to subject facet search based on hierarchy that includes v, x, y and z' do
+        stub_const 'SEPARATOR', '—'
```
Comment (banukutlu) [inline spec/helpers/catalog_helper_spec.rb:37]: 👌

**Pull request merged**